sky/resources.py

```diff
@@ -931,9 +931,20 @@ def less_demanding_than(self,
             if other.disk_tier is None:
                 return False
             if self.disk_tier != other.disk_tier:
+                if self.cloud is not None:
+                    self_tier = self.cloud.normalize_disk_tier(self.disk_tier)
+                else:
+                    self_tier = self.disk_tier
+                if other.cloud is not None:
+                    other_tier = other.cloud.normalize_disk_tier(
+                        other.disk_tier)
+                else:
+                    other_tier = other.disk_tier
+                # best disk tier without knowledge of cloud cannot be compared
```

Comment:
I thought it's guaranteed the self and other cloud are the same from Line 879?
so we don't need to translate it? the below code should be enough?